openml/datasets/dataset.py

```diff
@@ -311,6 +312,7 @@ def _download_data(self) -> None:
         if self._minio_url is not None:
             self.parquet_file = _get_dataset_parquet(self)
 
+    @typing.no_type_check
```

Comment:
Why do we need this?

Reply:
Hm, but we would basically know what type of data we have when applying these functions. Therefore, I would assume that it's possible to be more strict here. But let's leave this for now and focus on getting the rest done :)